### PR TITLE
If $on is a class, we will try to fetch the table from that class

### PR DESCRIPTION
**Pull request by jyrkidn**
This syntax is currently generating a wrong migration

```yaml
models:
  Event:
    working_title: string
    image_id: uuid foreign:\Foo\Bar\Attachment
```

```php
$table->foreign('image_id')->references('id')->on('\Foo\Bar\Attachments');
```

This makes that inserting record in mysql will fail, since `\Foo\Bar\Attachments` is not an existing table.

The changes in this pull request will fix this.

New output will be 
```php
$table->foreignUuid('image_id')->constrained('attachments');
```

### CODE REVIEW

**Comment by jasonmccreary** [comment]
@jyrkidn, can you update the PR details with the "fixed" output...

**Comment by jyrkidn** [comment]
Fixed output has been added and tests have been updated after the changes on master.

File: src/Generators/MigrationGenerator.php [hunk 1]
```diff
@@ -294,6 +294,9 @@ protected function buildForeignKey(string $column_name, ?string $on, string $typ
         } elseif (Str::contains($on, '.')) {
             [$table, $column] = explode('.', $on);
             $table = Str::snake($table);
+        } elseif (Str::contains($on, '\\')) {
+            $table = (new $on())->getTable();
```
Comment (jasonmccreary) [inline src/Generators/MigrationGenerator.php:298]: Depending on the order of operations (i.e. migrations generated before models), it is possible this model will not exist and as such this dynamic creation will fail.

**Review by jasonmccreary — CHANGES_REQUESTED** [review]
I'm wondering if this approach works in all cases...

File: src/Generators/MigrationGenerator.php (hunk 1 above)
Comment (jyrkidn) [reply src/Generators/MigrationGenerator.php:298]: Thought about that as well, first had a `class_exists` check. Maybe if the class does not exists, we can fallback to a plural, lowercase version of the model? E.g. `\Foo\Bar\Attachment` becomes `attachments`,

Comment (jasonmccreary) [reply src/Generators/MigrationGenerator.php:298]: To limit the code paths, I would not use the dynamic call and just do the plural, lowercase name of the model. I believe this is done elsewhere in the code.

Comment (jyrkidn) [reply src/Generators/MigrationGenerator.php]: Updated!

**Comment by jasonmccreary** [comment]
Awesome. Thanks!

**Comment by jyrkidn** [comment]
Thanks for reviewing and merging my pull requests. Very much appreciated!

**Pull request merged**